python/fusion_engine_client/analysis/analyzer.py

```diff
@@ -128,7 +128,7 @@ def __init__(self,
                @ref LONG_LOG_DURATION_SEC).
         """
         if isinstance(file, str):
-            self.reader = DataLoader(file, ignore_index=ignore_index)
+            self.reader = DataLoader(file, ignore_index=ignore_index, source_ids=source_id)
```

Comment:
I'd suggest a few changes here:
- Change the param name above to `source_ids` for consistency with other files
- Set the default to `None` (and make the type `Optional[List[int]]`) -- the default behavior should be "load all available data", and should also not assume that 0 is an available ID (even though it typically is)
- Don't specify `source_ids` in the `DataLoader` constructor (in fact as noted in that file, remove this argument)
- Remove the whole `default_source_id` business. The intent of this argument is really for the user to specify which pose source they're interested in. If they do `--source-id=1`, there's no reason for us to drop all event notifications for example